Index: firewall
===================================================================
--- firewall	(revision 8390)
+++ firewall	(working copy)
@@ -486,9 +486,12 @@
 [ $# -gt 1 ] && [ "$1" = "nolock" ] && { NOLOCK=Yes; shift ; }
 
 SHAREDIR=/usr/share/shorewall
-VARDIR=/var/lib/shorewall
 CONFDIR=/etc/shorewall
 
+[ -f ${CONFDIR}/vardir ] && . ${CONFDIR}/vardir ]
+
+[ -n "${VARDIR:=/var/lib/shorewall}" ]
+
 for library in lib.base lib.config; do
     FUNCTIONS=${SHAREDIR}/${library}
 
