diff -Naur -X /usr/local/bin/exclude.txt shorewall-perl-4.0.8/Shorewall/Rules.pm shorewall-perl-4.0.8.1/Shorewall/Rules.pm
--- shorewall-perl-4.0.8/Shorewall/Rules.pm	2008-01-22 15:55:50.000000000 -0800
+++ Shorewall/Rules.pm	2008-02-03 10:43:54.000000000 -0800
@@ -848,11 +848,12 @@
 	fatal_error "Invalid Action ($mtarget) in macro" unless $actiontype & ( ACTION +  STANDARD + NATRULE + MACRO );
 
 	if ( $msource ) {
-	    if ( ( $msource eq '-' ) || ( $msource eq 'SOURCE' ) ) {
+	    if ( $msource eq '-' ) {
 		$msource = $source || '';
-	    } elsif ( $msource eq 'DEST' ) {
-		$msource = $dest || '';
+	    } elsif ( $msource =~ s/^DEST:?// ) {
+		$msource = merge_macro_source_dest $msource, $dest; 
 	    } else {
+		$msource =~ s/^SOURCE:?//;
 		$msource = merge_macro_source_dest $msource, $source;
 	    }
 	} else {
@@ -860,11 +861,12 @@
 	}
 
 	if ( $mdest ) {
-	    if ( ( $mdest eq '-' ) || ( $mdest eq 'DEST' ) ) {
+	    if ( $mdest eq '-' ) {
 		$mdest = $dest || '';
-	    } elsif ( $mdest eq 'SOURCE' ) {
-		$mdest = $source || '';
+	    } elsif ( $mdest =~ s/^SOURCE:?// ) {
+		$mdest = merge_macro_source_dest $mdest , $source;
 	    } else {
+		$mdest =~ s/DEST:?//;
 		$mdest = merge_macro_source_dest $mdest, $dest;
 	    }
 	} else {
