Index: Shorewall/Config.pm
===================================================================
--- Shorewall/Config.pm	(revision 7800)
+++ Shorewall/Config.pm	(working copy)
@@ -1145,25 +1145,25 @@
     }
 }
 
-my %validlevels = ( debug   => 7,
-		    info    => 6,
-		    notice  => 5,
-		    warning => 4,
-		    warn    => 4,
-		    err     => 3,
-		    error   => 3,
-		    crit    => 2,
-		    alert   => 1,
-		    emerg   => 0,
-		    panic   => 0,
-		    none    => '',
+my %validlevels = ( DEBUG   => 7,
+		    INFO    => 6,
+		    NOTICE  => 5,
+		    WARNING => 4,
+		    WARN    => 4,
+		    ERR     => 3,
+		    ERROR   => 3,
+		    CRIT    => 2,
+		    ALERT   => 1,
+		    EMERG   => 0,
+		    PANIC   => 0,
+		    NONE    => '',
 		    ULOG    => 'ULOG' );
 
 #
 # Validate a log level -- Drop the trailing '!' and translate to numeric value if appropriate"
 #
 sub validate_level( $ ) {
-    my $level = $_[0];
+    my $level = uc $_[0];
 
     if ( defined $level && $level ne '' ) {
 	$level =~ s/!$//;
