Index: Shorewall/Zones.pm
===================================================================
--- Shorewall/Zones.pm	(revision 8219)
+++ Shorewall/Zones.pm	(working copy)
@@ -623,7 +623,7 @@
 	if ( defined $port ) {
 	    fatal_error qq("Virtual" interfaces are not supported -- see http://www.shorewall.net/Shorewall_and_Aliased_Interfaces.html) if $port =~ /^\d+$/;
 	    require_capability( 'PHYSDEV_MATCH', 'Bridge Ports', '');
-	    fatal_error "Your iptables is not recent enough to support bridge ports" unless $config{KLUDGEFREE};
+	    fatal_error "Your iptables is not recent enough to support bridge ports" unless $capabilities{KLUDGEFREE};
 	    fatal_error "Duplicate Interface ($port)" if $interfaces{$port};
 	    fatal_error "$interface is not a defined bridge" unless $interfaces{$interface} && $interfaces{$interface}{options}{bridge};
 	    fatal_error "Bridge Ports may only be associated with 'bport' zones" if $zone && $zoneref->{type} ne 'bport4';
