--- tags/4.0.6/Shorewall-perl-3/Shorewall/Nat.pm	2007/12/08 17:41:29	7863
+++ tags/4.0.6/Shorewall-perl-3/Shorewall/Nat.pm	2007/12/08 17:50:05	7864
@@ -362,14 +362,11 @@
     fatal_error "Invalid nat file entry" unless defined $interface && defined $internal;
 
     if ( $add_ip_aliases ) {
-	if ( $interface =~ s/:$// ) {
+	if ( defined( $alias ) && $alias eq '' ) {
 	    $add_ip_aliases = '';
 	} else {
-	    my ( $iface , undef ) = split /:/, $interface;
-	    emit "del_ip_addr $external $iface" unless $config{RETAIN_ALIASES};
+	    emit "del_ip_addr $external $interface" unless $config{RETAIN_ALIASES};
 	}
-    } else {
-	$interface =~ s/:$//;
     }
 
     validate_nat_column 'ALL INTERFACES', \$allints;
