diff -up ./shorewallrc.redhat.install ./shorewallrc.redhat
--- ./shorewallrc.redhat.install	2012-04-10 08:04:24.000000000 -0600
+++ ./shorewallrc.redhat	2012-04-11 15:17:01.214739928 -0600
@@ -5,8 +5,8 @@ BUILD=
 HOST=redhat
 PREFIX=/usr
 SHAREDIR=${PREFIX}/share
-LIBEXECDIR=${PREFIX}/share
-PERLLIBDIR=/usr/share/shorewall
+LIBEXECDIR=${PREFIX}/libexec
+PERLLIBDIR=${PREFIX}/share/perl5
 CONFDIR=/etc
 SBINDIR=/sbin
 MANDIR=${SHAREDIR}/man
