diff -up shorewall-lite-4.4.23.3/shorewall-lite.service.orig shorewall-lite-4.4.23.3/shorewall-lite.service
--- shorewall-lite-4.4.23.3/shorewall-lite.service.orig	2011-10-10 22:50:26.385507446 +0100
+++ shorewall-lite-4.4.23.3/shorewall-lite.service	2011-10-10 22:50:42.615956249 +0100
@@ -14,7 +14,6 @@ RemainAfterExit=yes
 EnvironmentFile=-/etc/sysconfig/shorewall-lite
 StandardOutput=syslog
 ExecStart=/sbin/shorewall-lite $OPTIONS start
-ExecReload=/sbin/shorewall-lite $OPTIONS restart
 ExecStop=/sbin/shorewall-lite $OPTIONS stop
 
 [Install]
