diff -up shorewall6-4.4.23.3/shorewall6.service.orig shorewall6-4.4.23.3/shorewall6.service
--- shorewall6-4.4.23.3/shorewall6.service.orig	2011-10-10 22:51:02.871268507 +0100
+++ shorewall6-4.4.23.3/shorewall6.service	2011-10-10 22:51:21.918621930 +0100
@@ -14,7 +14,6 @@ RemainAfterExit=yes
 EnvironmentFile=-/etc/sysconfig/shorewall6
 StandardOutput=syslog
 ExecStart=/sbin/shorewall6 $OPTIONS start
-ExecReload=/sbin/shorewall6 $OPTIONS restart
 ExecStop=/sbin/shorewall6 $OPTIONS stop
 
 [Install]
