diff --git a/Shorewall/Perl/Shorewall/Rules.pm b/Shorewall/Perl/Shorewall/Rules.pm
index c8b0d15..c06c653 100644
--- a/Shorewall/Perl/Shorewall/Rules.pm
+++ b/Shorewall/Perl/Shorewall/Rules.pm
@@ -2252,7 +2252,7 @@ sub build_zone_list( $$$\$\$ ) {
     #
     # Handle Wildcards
     #
-    if ( $input =~ /^(all[-+]*)(![^:]+)?(:.*)?/ ) {
+    if ( $input =~ /^(all[-+]*)(![^:]+)?(:.*)?$/ ) {
 	$input   = $1;
 	$exclude = $2;
 	$rest    = $3;
